api/src/test/java/io/grpc/GlobalInterceptorsTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+public class GlobalInterceptorsTest {
+
+  private static final ServerStreamTracer.Factory DUMMY_SERVER_TRACER =
+      new ServerStreamTracer.Factory() {
+        @Override
+        public ServerStreamTracer newServerStreamTracer(String fullMethodName, Metadata headers) {
+          throw new UnsupportedOperationException();
+        }
+      };
+
+  @Test
+  public void setClientInterceptors() {
+    GlobalInterceptors instance = new GlobalInterceptors();
+    List<ClientInterceptor> clientInterceptors = new ArrayList<>();
+    clientInterceptors.add(new NoopClientInterceptor());
+    clientInterceptors.add(new NoopClientInterceptor());
+
+    instance.setGlobalInterceptorsTracers(clientInterceptors, null, null);
+    assertThat(instance.getGlobalClientInterceptors()).hasSize(2);
+  }
+
+  @Test
+  public void setServerInterceptors() {
+    GlobalInterceptors instance = new GlobalInterceptors();
+    List<ServerInterceptor> serverInterceptors = new ArrayList<>();
+    serverInterceptors.add(new NoopServerInterceptor());
+
+    instance.setGlobalInterceptorsTracers(null, serverInterceptors, null);
+    assertThat(instance.getGlobalServerInterceptors()).hasSize(1);
+  }
+
+  @Test
+  public void setServerStreamTracerFactories() {
+    GlobalInterceptors instance = new GlobalInterceptors();
+    List<ServerStreamTracer.Factory> serverStreamTracerFactories = new ArrayList<>();
+    serverStreamTracerFactories.add(DUMMY_SERVER_TRACER);
+
+    instance.setGlobalInterceptorsTracers(null, null, serverStreamTracerFactories);
+    assertThat(instance.getGlobalServerStreamTracerFactories()).hasSize(1);
+  }
+
+  @Test
+  public void setInterceptorsTracers() {
```

Comment:
I thought since `setGlobalInterceptorsTracers` can be used to set three different variables (interceptors and tracer) I added other 3 tests. Is that not required because its trivial?